src/connection.c

```diff
@@ -296,21 +296,25 @@ void set_current_server (create_connection_ctx_t *ctx)
   ctx->current_server = get_current_server (&ctx->server_list);
 }
 
-void set_extra_headers (create_connection_ctx_t *ctx, int reauthorize)
+void free_extra_headers (create_connection_ctx_t *ctx)
+{
+  FREE_PTR_VAR (ctx->extra_headers)
+}
+
+void set_extra_headers (create_connection_ctx_t *ctx)
 {
-  if (reauthorize && (get_parodus_cfg()->client_cert_path !=NULL && strlen(get_parodus_cfg()->client_cert_path) >0))
+  ParodusCfg * cfg = get_parodus_cfg();
+
+  free_extra_headers (ctx);
+  if ((strlen(cfg->webpa_auth_token) == 0) &&
+      (cfg->client_cert_path != NULL) && (strlen(cfg->client_cert_path) > 0))
   {
-    getAuthToken(get_parodus_cfg());
+    getAuthToken(cfg);
```

Comment:
This needs to be done only in 403 case and when the ((strlen(cfg->webpa_auth_token) == 0), not in any other case.
This is to avoid too many calls to auth server.